tests/conftest.py

```diff
@@ -395,18 +396,38 @@ def get_records2(driver, range):
         }
         return special_records2.get(driver, get_records(driver, range))
 
-    def test_equal(driver, val_in, val_out):
-        is_good = True
-        is_good = is_good and val_in['geometry'] == val_out['geometry']
-        for key in val_in['properties']:
-            if key in val_out['properties']:
-                if driver == 'FileGDB' and isinstance(val_in['properties'][key], int):
-                    is_good = is_good and str(val_in['properties'][key]) == str(int(val_out['properties'][key]))
-                else:
-                    is_good = is_good and str(val_in['properties'][key]) == str(val_out['properties'][key])
-            else:
-                is_good = False
-        return is_good
+    def _get_pos(driver, record):
+        if driver in {'BNA', 'DXF', 'GPX', 'GPSTrackMaker', 'PCIDSK'}:
+            return int(record['geometry']['coordinates'][1])
+        elif driver in {'DGN'}:
+            return int(record['geometry']['coordinates'][0][0])
+        else:
+            return int(record['properties']['position'])
+
+    def test_records(driver, true_data, test_data):
+        """ Test if unique elements of records are preserved"""
+
+        if not len(true_data) == len(test_data):
+            logging.error("Not the same number of records: {} != {}".format(len(true_data), len(test_data)))
+            return False
+
+        positions = set([_get_pos(driver, r) for r in true_data])
+        if not len(true_data) == len(positions):
+            logging.error("Could not extract all unique elements from true_data: {} != {}".format(len(true_data),
+                                                                                                  len(positions)))
+            return False
+
+        if not len(set([_get_pos(driver, r) for r in test_data])) == len(positions):
+            logging.error("Could not extract unique elements from test_data.")
+            return False
+
+        for test_record in test_data:
+            pos = _get_pos(driver, test_record)
+            if pos not in positions:
+                logging.error("Test record {} corresponds not to a record from true_data.".format(test_record))
+                return False
+
+        return True
```

Comment:
You should just `assert` these things and let pytest print out the parts that don't match.

Reply:
The assert is in the corresponding tests.

Reply:
It's less effective; the tests will just `assert False` and that's all pytest will show. Then you have to go through the log to figure out what from here failed. If you assert here, then pytest will show the individual parts of what are not matching, and the backtrace will point to the actual failure.

Reply:
I agree with @QuLogic. We should let pytest do the work for us.

```python
assert len(actual_data) == len(expected_data)
```

Gives:

```
    def test_me():
        expected_data = [1,2,3]
        actual_data = [1,2]
>       assert len(actual_data) == len(expected_data)
E       assert 2 == 3
E        +  where 2 = len([1, 2])
E        +  and   3 = len([1, 2, 3])

tests/test_me.py:4: AssertionError
```